unionfs/unionfs.go

```diff
@@ -712,7 +712,12 @@ func (fs *unionFS) GetXAttr(name string, attr string, context *fuse.Context) ([]
 	}
 	r := fs.getBranch(name)
 	if r.branch >= 0 {
-		return fs.fileSystems[r.branch].GetXAttr(name, attr, context)
+		value, code := fs.fileSystems[r.branch].GetXAttr(name, attr, context)
+		// We intercept ENOSYS and replace it with ENODATA; returning ENOSYS would cause FUSE to never to GetXAttr() again.
+		if code == fuse.ENOSYS {
+			return nil, fuse.ENODATA
+		}
+		return value, code
```

Comment:
I think this makes more sense in unionfs. Typically a filesystem will either implement GetXAttr or they'll return ENOSYS in which case the existing behavior is desirable. I don't think you want to force users to know they should set DisableXAttr to avoid a bunch of repeated useless requests.

Can you imagine a scenario where what you're describing would be beneficial? This issue would seem to be one with unionfs not one that applies more generally.